### PR TITLE
Handle 403, 404

### PR DESCRIPTION
**Pull request by logangrado**
Update handling of 400 error codes such that:

* 403: Handle the same way as 401
* 404:
  * Attempt to find credentials from normal sources
  * Do not prompt if credentials are not found
* not_prompting:
  * Allow discovery of credentials when not prompting  

Addresses #69

### CODE REVIEW

File: src/unearth/auth.py [hunk 1]
```diff
@@ -340,11 +341,11 @@ def handle_401(self, resp: Response, **kwargs: Any) -> Response:
...
 
-    def warn_on_401(self, resp: Response, **kwargs: Any) -> None:
+    def warn_on_401_403_404(self, resp: Response, **kwargs: Any) -> None:
         """Response callback to warn about incorrect credentials."""
-        if resp.status_code == 401:
+        if resp.status_code in [401, 403, 404]:
             logger.warning(
-                "401 Error, Credentials not correct for %s",
+                f"{resp.status_code} Error, Credentials not correct for %s",
                 resp.request.url,
```
Comment (frostming) [inline src/unearth/auth.py:349]:
```suggestion
                "%s Error, Credentials not correct for %s",
                resp.status_code,
                resp.request.url,
```

**Comment by logangrado** [comment]
@frostming Thanks for accepting this contribution.

Any idea how long it will be until this is picked up in `PDM`?

**Comment by frostming** [comment]
0.11.0 is released, you can get this feature by upgrading unearth.

**Comment by frostming** [comment]
It seems this causes a regression on some indexes that return 403 on not found. For example: https://download.pytorch.org/whl/numba/

When it fails to get that package on not found, it returns a 403 instead of 404, so that it will prompt for username and password.

The situation is complicated because some index return 403 on 401, while some return 403 on 404, that's quite unfortunate.

@logangrado

**Pull request merged**